cli/main.go

```diff
@@ -2,69 +2,16 @@ package main
 
 import (
 	"fmt"
-	"io"
 	"os"
-	"strings"
 
 	"github.com/gluster/glusterd2/cli/cmd"
-	log "github.com/sirupsen/logrus"
 )
 
-const (
-	defaultLogFile  = "./cli.log"
-	defaultLogLevel = "INFO"
-)
-
-var logWriter io.WriteCloser
-
-func openLogFile(filepath string) (io.WriteCloser, error) {
-	f, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
-	if err != nil {
-		return nil, err
-	}
-	return f, nil
-}
-
-func setLogOutput(w io.Writer) {
-	log.SetOutput(w)
-}
-
-func initLog(logFilePath string, logLevel string) error {
-	// Close the previously opened Log file
-	if logWriter != nil {
-		logWriter.Close()
-		logWriter = nil
-	}
-
-	l, err := log.ParseLevel(strings.ToLower(logLevel))
-	if err != nil {
-		setLogOutput(os.Stderr)
-		log.WithError(err).Debug("Failed to parse log level")
-		return err
-	}
-	log.SetLevel(l)
-	log.SetFormatter(&log.TextFormatter{FullTimestamp: true})
-
-	logFile, err := openLogFile(logFilePath)
-	if err != nil {
-		setLogOutput(os.Stderr)
-		log.WithError(err).Debug("Failed to open log file %s", logFilePath)
-		return err
-	}
-	setLogOutput(logFile)
-	logWriter = logFile
-	return nil
-}
 func main() {
-	err := initLog(defaultLogFile, defaultLogLevel)
```

Comment:
I'd like it for logging intialization to remain here. Log should be intialized as early as possible, so that we can catch all logs. Doing this would allow logs for `argsMigrate()` as well.

I know trying to do this with the cobra package will not be easy, as it seems that the first level of flags can be added only to the root command. But can you explore if it is possible to add and make use of flags before a root command is created?

If it isn't possible, we can merge this as it is now.

Reply:
I feel we can skip that part now since we can print the errors to stderr(or as user friendly error message) in case we need to log something in argsMigrate.

I think adding separate parsing only for these two fields may not add value. Let me know your thoughts.

Reply:
There need not be separate parsing just for these two flags. The flags can be defined in RootCmd, but we could make use of it here.
I'm consolidating the logging intialization into a single package with my refactorings. I can take a shot at doing this with it. I'm okay with merging this for now.

@kmlmhnn fix @prashanthpai's comment, and we can get this merged.